src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -556,7 +556,7 @@ public void Put(TStructuralType original)
         /// Update the instance object properties with the properties from the @odata.id object.
         /// </summary>
         /// <param name="original">The @odata.id object.</param>
-        internal void UpdateODataIdObject(TStructuralType original)
+        internal void UpdateUnchangedPropertiesInInstanceObject(TStructuralType original)
```

Comment:
why do we update the unchanged properties? We should only update the changed properties?

Reply:
Unfortunately we write the unchanged properties in the response by writing their default value e.g `string` property we write `null`, `int` property we write `0`. This makes the response look messy.

If we have a request below
PATCH /Customers
```json
{
    "@odata.context":"http://localhost:6285/odata/$metadata#Customers/$delta",
    "value":[
        {
            "@odata.type":"CustomersApiEF.Models.Customer",
            "Id":1,
            "Name":"Customer1",
            "Orders@odata.delta":[
                {
                    "@odata.id":"Customers(3)/Orders(1005)",
                    "Quantity": 1005
                },
                {
                    "Id": 2000,
                    "Price":200,
                    "Quantity":90
                }
            ]
        }
    ]
}
```

The response would be
```json
{
    "@odata.context":"context url",
    "value":[
        {
            "Id":1,
            "Name":"Customer1",
            "Age": 0,
            "Orders":[
                {
                    "Id":1005,
                    "Price": 0,
                    "Quantity": 1005
                },
                {
                    "Id": 2000,
                    "Price":200,
                    "Quantity":90
                }
            ]
        }
    ]
}
```

If you look at the response, you will see the Customer Age is 0. Since Age was not in the payload, we write the default value in the response. Same thing is happening in Order 1005.

What the method `UpdateUnchangedPropertiesInInstanceObject` is doing is to extract the values from the object in datastorage and update the unchanged properties in the `_instance` object of Delta<T> object. Hence in the response above, we will write the actual Age of the customer.